### DISCUSSION TITLE
Is if there is token distribution cheatcode in foundry

### DISCUSSION THREAD

**Discussion by 1anyway** [foundry-rs/foundry#4926] [Ideas]
Like this one https://arbiscan.io/token/0x912CE59144191C1204E64559FE8253a0e49E6548#balances
For an ERC20 token,  if there is a cheatcode `vm.distribution(token)` returning the holders, then it would be very handy.

**Comment by mds1** [comment]
There is not, and probably never will be—determining all the holders of a token is "hard" in the sense that forge can't just read the list from contract storage. Instead you have to fetch all ERC20 `Transfer` events from that token and parse them to see which addresses that received transfers at any point still have those tokens. If you need this list in your tests, your best bet is to find an API that can return the list and use `vm.ffi` to call that API. Alternatively, you can fetch the list at one point using an API or other means, and hardcode the results

**Comment by 1anyway** [comment]
All right, thank you for your reply.
